data/data-pipeline/data_pipeline/etl/score/tests/conftest.py

```diff
@@ -0,0 +1,118 @@
+import os
+from importlib import reload
+from pathlib import Path
+
+import pandas as pd
+import pytest
+from data_pipeline import config
+from data_pipeline.etl.score import etl_score_post, tests
+from data_pipeline.etl.score.etl_score_post import PostScoreETL
+
+
+def pytest_configure():
+    pytest.SNAPSHOT_DIR = Path(__file__).parent / "snapshots"
+
+
+@pytest.fixture(scope="session")
+def root(tmp_path_factory):
+    basetemp = Path.cwd() / "temp_dir"
+    os.environ["PYTEST_DEBUG_TEMPROOT"] = str(
+        basetemp
+    )  # this sets the location of the temp directory inside the project folder
+    basetemp.mkdir(parents=True, exist_ok=True)
+    root = tmp_path_factory.mktemp("root", numbered=False)
+    return root
+
+
+@pytest.fixture(autouse=True)
+def settings_override(monkeypatch, root):
```

Comment:
Rather than monkey-patching, how about setting dir/path/file names as env variables and having an `.env.test` that sets the path to the tmp path? Then we wouldn't need a lot of this code and we'd have flexibility for changing path names later or in other envs.

Reply:
Sounds like maybe this is python/pytest convention, so I don't have a strong opinion here, although I do still prefer env variables over monkeypatching to set attrs that are env-related ^.^

Reply:
Ideally this is done with dynaconf and env settings. Right now that's not working because we're building these values from variables or constants instead of setting them in settings.toml. We should try setting these in settings.toml. We'll do this in a future test PR (since this works as is and isn't too scary ^.^).

Note that we're calling `reload` in these methods too and that's something we shouldn't have to do (it's hacky!).